lib/absinthe_auth/middleware.ex

```diff
@@ -4,16 +4,38 @@ defmodule AbsintheAuth.Middleware do
   @behaviour Absinthe.Plugin
 
   def call(%{private: %{authorisation: :done}} = resolution, _) do
-    maybe_append_middleware(resolution)
+    resolution
   end
-  def call(resolution, {module, args}) do
+
+  # authorisation pending - call the policy module and check the result
+  def call(%{private: %{authorisation: :pending}} = resolution, {module, args}) do
     resolution
     |> module.call(args)
     |> maybe_continue_authorisation
   end
 
+  # No authorisation - init auth state to :pending, install middleware to reset after resolution
```

Comment:
Testing with Absinthe 1.5 I found that the `private` field is no longer reset automatically between fields. This was causing authorisations on earlier fields to leak into later fields on the schema.

This version makes it the responsibility of the first `AbsintheAuth.Middleware.call` on a field to install a middleware at the end of the pipeline to clear delete the `private.authorization` state on the resolution.